core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/Checksummer.java

```diff
@@ -35,12 +48,50 @@ static Checksummer create() {
     }
 
     /**
-     * Given a payload, calculate a checksum and return it as a string.
+     * Get a flexible checksummer that performs two checksums: the given checksum-algorithm and the default (sha256).
+     */
+    static Checksummer create(ChecksumAlgorithm checksumAlgorithm) {
```

Comment:
Suggesting `static Checksummer forFlexibleChecksum(ChecksumAlgorithm checksumAlgorithm)` and requiring checksumAlgorithims to be not null.